pos_order_to_sale_order/static/src/js/pos_order_to_sale_order.js

```diff
@@ -0,0 +1,137 @@
+/******************************************************************************
+    Copyright (C) 2017 - Today: GRAP (http://www.grap.coop)
+    @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+ *****************************************************************************/
+
+openerp.pos_order_to_sale_order = function(instance, local) {
+
+    "use strict";
+
+    var module = instance.point_of_sale;
+    var _t = instance.web._t;
+
+    /*************************************************************************
+        New Widget CreateSaleOrderButtonWidget:
+            * On click, check if there is a customer defined,
+              ask confirmation call server to create sale order, and delete
+              the current order. 
+    */
+    module.CreateSaleOrderButtonWidget = module.PosBaseWidget.extend({
+        template: 'CreateSaleOrderButtonWidget',
+
+        init: function(parent, options){
+            this._super(parent, options);
+            this.display_text = _t("Create Draft Order");
+            if (this.pos.config.iface_create_sale_order_action == 'confirmed_order'){
+                this.display_text = _t("Create Confirmed Order");
+            }
+            if (this.pos.config.iface_create_sale_order_action == 'delivered_picking'){
+                this.display_text = _t("Create Delivered Picking");
+            }
+        },
+
+        renderElement: function() {
+            var self = this;
+            this._super();
+            this.$el.click(function(){
+                var current_order = self.pos.get('selectedOrder');
+                // Prevent empty delivery order
+                if (current_order.get('orderLines').length == 0){
+                    self.pos_widget.screen_selector.show_popup('error',{
+                        'message': _t('Empty Order'),
+                        'comment': _t('There must be at least one product in your order to create Sale Order.'),
+                    });
+                    return;
+                }
+                // Check Customer
+                if (current_order.get('client') ===  null){
```

Comment:
Well, it works yes. Do you want I test with Undefined, because it's a best practice ?